### ISSUE TITLE
Crash in GRDB.DatabasePool.writer

### ISSUE THREAD

**Issue by nrkrishna** [groue/GRDB.swift#1035]
We have an app that uses GRDB to open a database in pool mode. I'm seeing several crashes in Firebase Crashlytics with a crash trace as pasted below. I'm not sure where to start to debug why this might be happening. As context for how we are using GRDB, we have value observers that are listening to updates to various tables whilst some of those same tables are also being written to. The writes happen infrequently, such as when the app is starting up. I'm hoping to get some advice on how I can go back identifying the cause of the crash. Unfortunately, I'm unable to reproduce them at my end - so the symbolicated crash logs is all I have. Any tips/advice massively appreciated? 

**GRDB flavor(s):** GRDB
**GRDB version:** 5.8.0
**Installation method:** CocoaPods
**Xcode version:** 12.5.1
**Swift version:** 5.0
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** 11.0.1

```
Crashed: GRDB.DatabasePool.writer
0  libswiftCore.dylib             0x1a69b09a0 swift_retain + 60
1  libswiftCore.dylib             0x1a69ecd60 swift_bridgeObjectRetain + 56
2  libswiftCore.dylib             0x1a69b43c4 swift::metadataimpl::ValueWitnesses<swift::metadataimpl::BridgeObjectBox>::initializeWithCopy(swift::OpaqueValue*, swift::OpaqueValue*, swift::TargetMetadata<swift::InProcess> const*) + 28
3  GRDB                           0x10201ba14 ValueObserver.recordingSelectedRegionIfNeeded<A>(_:fetch:) + 217 (ValueObserver.swift:217)
4  GRDB                           0x10201b400 ValueObserver.databaseDidCommit(_:) + 95 (ValueObserver.swift:95)
5  GRDB                           0x10200f2c4 DatabaseObservationBroker.databaseDidCommit() + 854 (TransactionObserver.swift:854)
6  GRDB                           0x10200dfc0 DatabaseObservationBroker.updateStatementDidExecute(_:) + 391 (TransactionObserver.swift:391)
7  GRDB                           0x101eca208 Database.executeUpdateStatement(_:) + 334 (Database+Statements.swift:334)
8  GRDB                           0x101feb94c UpdateStatement.execute(arguments:) + 544 (Statement.swift:544)
9  GRDB                           0x101ec9eb4 Database.execute(literal:) + 255 (Database+Statements.swift:255)
10 GRDB                           0x101ed17a8 Database.inTransaction(_:_:) + 185 (Database+Statements.swift:185)
11 GRDB                           0x101f0fa3c closure #1 in DatabaseWriter.asyncWrite<A>(_:completion:) + 268 (DatabaseWriter.swift:268)
12 GRDB                           0x101f950fc partial apply for closure #1 in SerializedDatabase.async(_:) + 192 (SerializedDatabase.swift:192)
13 GRDB                           0x101ee70fc thunk for @escaping @callee_guaranteed () -> () + 4355715324 (<compiler-generated>:4355715324)
14 libdispatch.dylib              0x1a26b1a84 _dispatch_call_block_and_release + 32
15 libdispatch.dylib              0x1a26b381c _dispatch_client_callout + 20
16 libdispatch.dylib              0x1a26bb004 _dispatch_lane_serial_drain + 620
17 libdispatch.dylib              0x1a26bbc00 _dispatch_lane_invoke + 404
18 libdispatch.dylib              0x1a26c64bc _dispatch_workloop_worker_thread + 764
19 libsystem_pthread.dylib        0x1ee7367a4 _pthread_wqthread + 276
20 libsystem_pthread.dylib        0x1ee73d74c start_wqthread + 8
```

**Comment by groue** [comment]
Hello @nrkrishna,

I'm sorry for the crash. 

This looks like #1026, which was fixed in v5.9.0. Please upgrade, and open a new issue if the issue persists.

**Comment by nrkrishna** [comment]
Thanks @groue for the quick response. I will do that and get back if I see this happening with v5.9.0. Much appreciated!
